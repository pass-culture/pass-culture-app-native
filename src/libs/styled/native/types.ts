import { DefaultTheme } from 'styled-components/native'

export type Computed = {
  isMobileViewport?: boolean
  isTabletViewport?: boolean
  isDesktopViewport?: boolean
  showTabBar: boolean
  appContentWidth: number
}

export type ComputedTheme = DefaultTheme & Computed
