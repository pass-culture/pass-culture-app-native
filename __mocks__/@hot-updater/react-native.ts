import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
