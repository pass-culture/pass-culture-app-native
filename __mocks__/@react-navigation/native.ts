export { getStateFromPath, getPathFromState } from '@react-navigation/native'
import { useEffect } from 'react'

export const addListener = jest.fn()
export const removeListener = jest.fn()
export const navigate = jest.fn()
export const replace = jest.fn()
export const reset = jest.fn()
export const goBack = jest.fn()
export const canGoBack = jest.fn()

const navigation = {
  addListener,
  removeListener,
  navigate,
  replace,
  reset,
  goBack,
  canGoBack,
}
export const useNavigation = () => navigation
export const createNavigationContainerRef = () => ({
  ...navigation,
  isReady: () => true,
  current: navigation,
})

export const useRoute = jest.fn()
export const useFocusEffect = useEffect
export const NavigationContainer = jest.fn()
export const useNavigationState = jest.fn()
