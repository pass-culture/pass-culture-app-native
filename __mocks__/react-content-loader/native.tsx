import React, { ReactNode } from 'react'
import { Svg } from 'react-native-svg'

const MockContentLoader = ({ children }: { children: ReactNode }) => {
  return <Svg>{children}</Svg>
}
export default MockContentLoader
