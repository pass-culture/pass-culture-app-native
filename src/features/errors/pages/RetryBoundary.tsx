import { t } from '@lingui/macro'
import React from 'react'
import { FallbackProps } from 'react-error-boundary'
import { useQueryErrorResetBoundary } from 'react-query'
import styled from 'styled-components/native'

import { _ } from 'libs/i18n'
import { AppButton } from 'ui/components/buttons/AppButton'
import { Background } from 'ui/svg/Background'
import { BrokenConnection } from 'ui/svg/BrokenConnection'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'

export const RetryBoundary = ({ resetErrorBoundary }: FallbackProps) => {
  const { reset } = useQueryErrorResetBoundary()

  const handleRetry = () => {
    reset()
    resetErrorBoundary()
  }

  return (
    <Container>
      <Background />
      <Spacer.Flex />

      <BrokenConnection />
      <Spacer.Column numberOfSpaces={2} />

      <Typo.Title1 color={ColorsEnum.WHITE}>{_(t`Oops !`)}</Typo.Title1>
      <Spacer.Column numberOfSpaces={4} />

      <Row>
        <TextContainer>
          <CenteredText>
            <Typo.Body color={ColorsEnum.WHITE}>
              {_(t`Une erreur s'est produite pendant le chargement.`)}
            </Typo.Body>
          </CenteredText>
        </TextContainer>
      </Row>

      <Spacer.Column numberOfSpaces={8} />

      <Row>
        <ButtonContainer>
          <AppButton
            title={_(t`RÃ©essayer`)}
            onPress={handleRetry}
            textColor={ColorsEnum.WHITE}
            borderColor={ColorsEnum.WHITE}
            loadingIconColor={ColorsEnum.WHITE}
            buttonHeight="tall"
          />
        </ButtonContainer>
      </Row>
      <Spacer.Flex />
    </Container>
  )
}

const Container = styled.View({
  flex: 1,
  alignItems: 'center',
})

const Row = styled.View({ flexDirection: 'row' })

const ButtonContainer = styled.View({ flex: 1, maxWidth: getSpacing(44) })
const TextContainer = styled.View({ maxWidth: getSpacing(88) })

const CenteredText = styled.Text({
  textAlign: 'center',
})
