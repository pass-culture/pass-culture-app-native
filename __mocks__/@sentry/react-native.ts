import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
