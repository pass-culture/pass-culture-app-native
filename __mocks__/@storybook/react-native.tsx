import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
