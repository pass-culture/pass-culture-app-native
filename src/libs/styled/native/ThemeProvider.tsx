import React from 'react'
import { DefaultTheme, ThemeProvider as DefaultThemeProvider } from 'styled-components/native'

import { useComputedTheme } from '../useComputedTheme'

export const ThemeProvider: React.FC<{ theme: DefaultTheme }> = ({ children, theme }) => {
  const computedTheme = useComputedTheme<DefaultTheme>(theme)
  return <DefaultThemeProvider theme={computedTheme}>{children}</DefaultThemeProvider>
}
