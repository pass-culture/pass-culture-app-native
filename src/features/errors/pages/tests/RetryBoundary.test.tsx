import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import { canGoBack, goBack } from '__mocks__/@react-navigation/native'

import { RetryBoundary } from '../RetryBoundary'

describe('RetryBoundary component', () => {
  it('should render', () => {
    const resetErrorBoundary = jest.fn()
    const component = render(
      <RetryBoundary error={new Error('error')} resetErrorBoundary={resetErrorBoundary} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should have back arrow if possible', () => {
    canGoBack.mockImplementation(() => true)
    const { getByTestId, queryByTestId } = render(
      <RetryBoundary error={new Error('error')} resetErrorBoundary={jest.fn()} />
    )
    expect(queryByTestId('backArrow')).toBeTruthy()
    fireEvent.press(getByTestId('backArrow'))
    expect(goBack).toHaveBeenCalled()
  })
  it('should not have back arrow if impossible', () => {
    canGoBack.mockImplementation(() => false)
    const { queryByTestId } = render(
      <RetryBoundary error={new Error('error')} resetErrorBoundary={jest.fn()} />
    )
    expect(queryByTestId('backArrow')).toBeFalsy()
  })
})
