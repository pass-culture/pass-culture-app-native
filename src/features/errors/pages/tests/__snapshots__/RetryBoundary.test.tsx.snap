// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetryBoundary component should render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="none"
      bbHeight="100%"
      bbWidth="100%"
      focusable={false}
      height="100%"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      vbHeight={667}
      vbWidth={375}
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1376171,
                1,
                -13500266,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="prefix__b"
            x1="65.805%"
            x2="47.731%"
            y1="26.588%"
            y2="116.28%"
          />
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1376171,
                1,
                -13500266,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="prefix__d"
            x1="29.678%"
            x2="32.571%"
            y1="71.761%"
            y2="14.036%"
          />
          <RNSVGPath
            d="M0 0h375v667H0z"
            name="prefix__a"
          />
          <RNSVGPath
            d="M574.848 685.018c44.64 237.462 197.943-58.73 172.765-186.83-25.178-128.1-120.317-395.218-387.196-385.031-266.88 10.187-281.873 242.282-174.83 360.46 107.043 118.179 344.621-26.06 389.26 211.401z"
            name="prefix__e"
          />
        </RNSVGDefs>
        <RNSVGGroup
          fill={null}
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        >
          <RNSVGMask
            fill={4294967295}
            height="100%"
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            maskUnits={0}
            name="prefix__c"
            propList={
              Array [
                "fill",
              ]
            }
            width="100%"
            x="0%"
            y="0%"
          >
            <RNSVGUse
              height={0}
              href="prefix__a"
              width={0}
              x={0}
              y={0}
            />
          </RNSVGMask>
          <RNSVGUse
            fill={
              Array [
                1,
                "prefix__b",
              ]
            }
            height={0}
            href="prefix__a"
            matrix={
              Array [
                -1,
                0,
                0,
                1,
                375,
                0,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            width={0}
            x={0}
            y={0}
          />
          <RNSVGGroup
            mask="prefix__c"
          >
            <RNSVGUse
              fill={
                Array [
                  1,
                  "prefix__d",
                ]
              }
              height={0}
              href="prefix__e"
              matrix={
                Array [
                  -0.8660254037844387,
                  -0.49999999999999994,
                  0.49999999999999994,
                  -0.8660254037844387,
                  247.47746608921344,
                  990.3613871835053,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="124"
    bbWidth="197"
    focusable={false}
    height="124"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 124,
          "width": 197,
        },
      ]
    }
    vbHeight={124}
    vbWidth={197}
    width="197"
  >
    <RNSVGGroup>
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M168.453 101.455c1.03 0 1.865.84 1.865 1.878 0 1.038-.835 1.88-1.865 1.88h-2.526c-1.03 0-1.864-.842-1.864-1.88 0-1.037.835-1.878 1.864-1.878zm-83.308-83.92c19.218 0 34.802 15.595 34.802 34.839 0 2.505-.265 4.976-.784 7.384l1.603.497c4.238 1.314 6.614 5.826 5.324 10.107l-4.955 16.441c-1.298 4.305-5.813 6.752-10.092 5.462l-1.046-.324v1.168c0 4.418 3.488 8.016 7.841 8.147l.248.004h3.415c4.634 0 8.39-3.785 8.39-8.455v-3.064c0-5.052 4.064-9.148 9.077-9.148 5.014 0 9.078 4.096 9.078 9.148v5.883c0 3.113 2.504 5.636 5.593 5.636h3.171c1.03 0 1.865.841 1.865 1.879 0 1.037-.835 1.879-1.865 1.879h-3.171c-5.149 0-9.322-4.206-9.322-9.394V89.74c0-2.977-2.395-5.39-5.349-5.39s-5.348 2.413-5.348 5.39v3.064c0 6.745-5.426 12.213-12.119 12.213h-3.415c-6.527 0-11.818-5.332-11.818-11.91v-2.324l-1.238-.384c-2.777-.861-4.756-3.097-5.425-5.741-.069-.01-.138-.024-.207-.04l-.15-.04c-1.636-.473-2.57-2.185-2.096-3.825l5.461-18.86c.475-1.642 2.174-2.597 3.807-2.14l.146.046c.194.065.378.148.55.246 1.982-1.728 4.753-2.455 7.448-1.711l.183.053.838.26c.42-2.048.633-4.146.633-6.274 0-17.163-13.909-31.081-31.073-31.081-17.163 0-31.072 13.918-31.072 31.08 0 2.064.2 4.1.596 6.09l.24-.075.184-.053c2.694-.744 5.466-.017 7.45 1.711.17-.098.353-.181.547-.246l.146-.046c1.633-.457 3.332.498 3.807 2.14l5.461 18.86c.475 1.64-.46 3.352-2.095 3.825l-.15.04c-.07.015-.139.03-.208.04-.67 2.644-2.647 4.88-5.425 5.741l-6.013 1.865c-4.279 1.29-8.794-1.157-10.092-5.462l-4.955-16.441c-1.29-4.281 1.087-8.793 5.324-10.107l2.198-.682c-.493-2.35-.744-4.76-.744-7.2 0-19.243 15.584-34.838 34.801-34.838zm23.087 47.393l-2.235 7.416-2.9 10.012c-.13 1.975 1.09 3.849 3.046 4.456l5.95 1.845.185.053c2.257.598 4.602-.713 5.287-2.983l4.955-16.441c.695-2.305-.585-4.735-2.867-5.442l-5.957-1.848-.177-.051c-2.257-.597-4.603.713-5.287 2.983zm-46.808 0c-.684-2.27-3.03-3.58-5.286-2.983l-.178.05-5.957 1.849c-2.282.707-3.561 3.137-2.867 5.442l4.956 16.441c.684 2.27 3.029 3.58 5.286 2.983l.185-.053 5.95-1.845c1.957-.608 3.177-2.483 3.046-4.459l-2.9-10.01zm109.11 6.784c.79-.672 1.969-.578 2.633.21.664.788.562 1.971-.227 2.643l-4.148 3.53 1.288.775c1.177.707 1.554 2.241.841 3.427-.141.235-.32.445-.527.622l-5.27 4.485c-.79.672-1.968.578-2.633-.21-.664-.788-.562-1.971.228-2.643l3.94-3.354-1.289-.774c-.233-.14-.44-.317-.616-.525-.886-1.05-.75-2.628.303-3.524zM24.438 14.052c.862-.565 2.025-.32 2.596.544l4.839 7.321 1.848-2.203c.883-1.051 2.457-1.183 3.517-.294.186.157.349.34.483.542l5.555 8.405c.571.865.336 2.024-.526 2.588-.862.564-2.024.32-2.596-.544l-4.627-7.003-1.849 2.203c-.155.185-.337.347-.539.479-1.149.752-2.699.427-3.461-.726l-5.765-8.724c-.572-.865-.336-2.024.525-2.588z"
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    numberOfSpaces={2}
    style={
      Array [
        Object {
          "height": 8,
        },
      ]
    }
  />
  <Text
    color="#ffffff"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Montserrat-ExtraBoldItalic",
          "fontSize": 28,
          "fontWeight": "800",
          "lineHeight": 34,
        },
      ]
    }
  >
    Oops !
  </Text>
  <View
    numberOfSpaces={4}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "maxWidth": 352,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        <Text
          color="#ffffff"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 15,
                "fontWeight": "normal",
                "lineHeight": 20,
              },
            ]
          }
        >
          Une erreur s'est produite pendant le chargement.
        </Text>
      </Text>
    </View>
  </View>
  <View
    numberOfSpaces={8}
    style={
      Array [
        Object {
          "height": 32,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "maxWidth": 176,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#ffffff",
            "borderRadius": 24,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "maxWidth": 500,
            "opacity": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
            "width": "100%",
          }
        }
        testID="button-container"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 5,
                "maxWidth": 650,
              },
            ]
          }
          testID="button-title"
          textColor="#ffffff"
        >
          Réessayer
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
